splink/dialects.py

```diff
@@ -80,6 +84,47 @@ def name(self):
     def levenshtein_function_name(self):
         return "levenshtein"
 
+    def date_diff(self, clc: "ComparisonLevelCreator"):
+        """Note some of these functions are not native postgres functions and
+        instead are UDFs which are automatically registered by Splink
+        """
+
+        if clc.date_format is None:
+            clc.date_format = "yyyy-MM-dd"
+
+        col_name_l = clc.input_column(self).name_l()
+        col_name_r = clc.input_column(self).name_r()
+
+        if clc.cast_strings_to_date:
+            datediff_args = f"""
+                to_date({col_name_l}, '{clc.date_format}'),
```

Comment:
I am saying this without checking if this is going to be used elsewhere, but we could have this baked into the `InputColumn` class.

Reply:
very interesting idea - also regex, lower, etc.